### PR TITLE
Update to SBML 5.16

### PR DESCRIPTION
**Pull request by tpfau**
Update to SBML 5.16
This update includes:
Corrected IO for subSystems (they are now exported via the groups plugin - in accordance with BiGG models). They can now also be read from those.
Fixed issues with macOS high Sierra/Sierra and current Matlab versions. 
SBML IO should work now with all current Matlab/macOS combinations indicated by mathworks.

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [] with [X] to check the box)*

### CODE REVIEW

File: src/base/io/writeCbModel.m [hunk 1]
```diff
@@ -179,6 +179,26 @@
     else
         return;
     end
+else
+    if strcmp(format,'toselect') % no format was given. try to detect from fileName.
+        [~, ~, extension] = fileparts(fileName);
+        switch extension
```
Comment (laurentheirendt) [inline src/base/io/writeCbModel.m]: May you put this `switch` in an internal function?

File: initCobraToolbox.m [hunk 2]
```diff
@@ -522,6 +527,48 @@ function initCobraToolbox()
     if ENV_VARS.printLevel
         clearvars
     end
+        
+    
+end
+
+
+function adaptMacPath()
```
Comment (laurentheirendt) [inline initCobraToolbox.m]: May you put this function as a separate function in `base`?

**Comment by tpfau** [comment]
Thanks for the comments. Is this what you wanted?

**Comment by tpfau** [comment]
So all tests pass, but codecov fails and thus the default?

**Comment by tpfau** [comment]
Close/Reopen to retrigger the CI

**Pull request merged**